rebasehelper/plugins/spec_hooks/replace_old_version.py

```diff
@@ -58,22 +61,54 @@ def _is_local_source(cls, line: str) -> bool:
         return not urllib.parse.urlparse(source).scheme
 
     @classmethod
-    def run(cls, spec_file, rebase_spec_file, **kwargs):
+    def _create_possible_replacements(cls, old: str, new: str, replacement: str) -> Dict[Pattern[str], str]:
+        """Creates possible subversion replacements.
+
+        Args:
+            old: Old version.
+            new: New Version.
+            replacement: Regex replacement pattern.
+
+        Returns:
+            Mapping of old version substring regex Pattern to replacement
+            for re.sub.
+
+        Example:
+            Subversions 1.2, 1.2.3 would be created from version 1.2.3.4.
+
+        """
+        split_version = old.split('.')
+        res = {}
+        # iterate backwards to go from longer to shorter subversions
+        for i in reversed(range(2, len(split_version))):
+            pattern = re.compile(r'([/\-\s]){}([/.\-\s])'.format(re.escape('.'.join(split_version[:i]))))
+            res[pattern] = replacement.format('.'.join(new.split('.')[:i]))
+        return res
+
+    @classmethod
+    def run(cls, spec_file: SpecFile, rebase_spec_file: SpecFile, **kwargs: Any):
         old_version = spec_file.get_version()
         new_version = rebase_spec_file.get_version()
         replace_with_macro = kwargs.get('replace_old_version_with_macro')
-        pattern = re.compile(r'([/\-\s]){}([/.\-\s])'.format(re.escape(old_version)))
+
+        replacement_pattern = r'\g<1>{}\g<2>'
+        subversion_patterns = cls._create_possible_replacements(old_version, new_version, replacement_pattern)
+
+        main_pattern = re.compile(r'([/\-\s]){}([/.\-\s])'.format(re.escape(old_version)))
+        replacement = replacement_pattern.format('%{version}' if replace_with_macro else new_version)
+
         for sec_name, section in rebase_spec_file.spec_content.sections:
             if sec_name.startswith('%changelog'):
                 continue
             for index, line in enumerate(section):
                 # special case Version tag to avoid mistakes when version format changes
                 # e.g. in a rebase from 2.5 to 2.5.1, version would be changed to 2.5.1.1
-                if cls._is_local_source(line) or line.startswith('Version:'):
+                if cls._is_local_source(line) or line.startswith('Version'):
```

Comment:
What's the reason for this?

Reply:
For consistency. I've checked the rest of the code and from what I've seen that colon is never used when looking for particular preamble tags.

Reply:
Ok.